pkg/limayaml/limayaml.go

```diff
@@ -31,6 +31,7 @@ type LimaYAML struct {
 	Env          map[string]string `yaml:"env,omitempty" json:"env,omitempty"`
 	DNS          []net.IP          `yaml:"dns,omitempty" json:"dns,omitempty"`
 	HostResolver HostResolver      `yaml:"hostResolver,omitempty" json:"hostResolver,omitempty"`
+	UserNet      UserNet           `yaml:"userNet,omitempty" json:"userNet,omitempty"`
```

Comment:
>  Seems confusing to have two network options both called "user"...
>
> https://wiki.qemu.org/Documentation/Networking#User_Networking_(SLIRP)

_Originally posted by @afbjorklund in https://github.com/lima-vm/lima/issues/1222#issuecomment-1443778603_

Reply:
Bikeshedding:

- `gvisor` (confusing because it does not use gvisor's process sandbox)
- `gvisor-tap-vsock` (confusing because it is not gvisor, nor tap, nor vsock)
- `netstack` (confusing because any networking can be called "net(work) stack"
- `user-v2` (confusing because there was no `user-v1`)
- `modern` (confusing because it can be eventually legacy)

Reply:
Actually it would be OK to both of them "user", and then toggle the slirp vs gvisor later ?

Just was afraid that we would have one called "user" and a different one called "usernet"...

Reply:
Am bit confused on this "two network options both called user"

As of now we don't have any configuration for slirp right?
And the one we provide now will be more a toggle, if enabled will use the latest or revert to the old modal ?

Reply:
It is called user in qemu, is what I meant. If there is an option called usernet in lima too, it is confusing.

Reply:
What about calling it "builtin"

Reply:
Or "multivm", "multi-vm", "vm-to-vm", "vm2vm", "v2v"

Reply:
To me the following suggestions looks good,
- user-v2 as this give support for giving further versions as well like user-v3 etc.
- builtin since usernet command is within limactl it kind of makes sense as well

Reply:
@afbjorklund @jandubois
WDYT?

Reply:
I don't know what happened with "user-v1", is that the normal one ? But "user-v2" seems complicated to me, as a yaml default. But maybe it needs versioning (as per above, "user-v3", "user-v4") otherwise "builtin" is simpler.

Reply:
> What about calling it "builtin"

My vote is for `builtin`